src/_guides/language/async-await.md

```diff
@@ -0,0 +1,344 @@
+
+To see this lifecycle at work consider what happens after you invoke the `getUserOrder` function:
+
+```
+String createOrderMessage () {
+  var order = getUserOrder(); 
+  return 'Your order is: $order';
+}
+
+Future<String> getUserOrder() {
+  // Imagine that this function is more complex and slow
+  return Future.delayed(Duration(seconds: 4), () => 'Large Latte'); 
+}
+
+main() {
+  print(createOrderMessage()); 
+}
+```
+
+`// Your order is: Instance of '_Future<String>'`
+
+* after 1 second has passed: the future is uncomplete - we cannot access the desired result yet
+
+`     Your order is: Instance of '_Future<String>'`
+
+* after 4 seconds have passed: the future completes, returning the desired result 
+
+`'Large Latte'`
+
+
+But how do you access this result that completes 4 seconds later? In addition to providing future objects, Dart also provides a clean and easy way to handle their results -- the `async` and `await` keywords. 
+
+Quick Review:
+
+* In Dart, the result of an asynchronous operation is represented by a  [Future](https://api.dartlang.org/stable/2.2.0/dart-async/Future-class.html)`<T>` object that produces a result of type `T`. For example,  [Future](https://api.dartlang.org/stable/2.3.0/dart-async/Future-class.html)`<String>` is the type signature for a future object that produces a string result. If a future produces a result that isn't a usable value, then the future's type is `Future<void>`. 
+* A future can be in one of two states: uncompleted, or completed. 
+* A completed future can either (1) complete with a result or (2) complete with an error
+* When a function that returns a future is invoked, two things happen:
+
+1. The function queues up work to be done and returns an uncompleted future.
+2. Later, when the operation is finished, the future completes with a value or with an error.
+
+
+## Working with Futures in Dart: async and await
+
+
+
+In the previous section you learned that, in Dart, future objects represent the results of asynchronous operations -- if a function does asynchronous work, it returns a future object. But how do we access the results of future objects when using asynchronous functions? 
+
+The `async` and `await` keywords provide a declarative way to (1) define asynchronous functions and (2) access their results  *after they have completed.* 
+
+
+To define an `async` function: 
+
+1. Add the `async` keyword after the function's parameters but before the function's body. 
+
+    createOrderMessage () async {
+
+2. Update the function's type signature to return a future. In the following example, we define an `async` function that completes with a string.   
+
+    Future<String> createOrderMessage () async {
+
+ *Note: You can *  [use Future<void>](https://dart.dev/guides/language/effective-dart/design#do-use-futurevoid-as-the-return-type-of-asynchronous-members-that-do-not-produce-values) * as the return type for asynchronous functions that don't return usable values. * 
```

Comment:
I did some digging and ended up with `<div class="alert alert-info" markdown="1">` but feel free to let me know if we have different pattern for this.

Reply:
No, we use a liquid template:

```
{{site.alert.info}}    (I think we have a .info..., otherwise site.alert.note or 
                                 site.alert.secondary, depending)
  Text here
{{site.alert.end}}
```

Reply:
We've configured these templates in flutter/website, but not in dart-lang/site-www. 

Notice that  after running `jekyll build`, the following :
```
{{ site.alert.info }}
INFO
{{ site.alert.end }}

{{ site.alert.note }}
NOTE
{{ site.alert.end }}

{{ site.alert.secondary }}
SECONDARY
{{ site.alert.end }}
```

renders to:

![Screen Shot 2019-06-05 at 4 51 30 PM](https://user-images.githubusercontent.com/5550175/58997640-5a465d00-87b2-11e9-8dcb-e9f4950d9e9a.png)

Reply:
I'll update the PR to add these templates to the site's config.yml, then sub out the divs for the templates.

Reply:
Quick update about these notes:

IMHO these styles don't match the use case I have in mind. I was going for something subtle that didn't draw too much of the the viewer's attention. These notes are really bulky, and after just 2 of them they start swallowing screen real estate and distract the viewer from the main flow of the article/tutorial/codelab:

### Before: 

![Screen Shot 2019-06-07 at 3 26 34 PM](https://user-images.githubusercontent.com/5550175/59136608-ba670b80-8938-11e9-9ce3-2ed06dc9b3be.png)

### After:
![Screen Shot 2019-06-07 at 3 19 29 PM](https://user-images.githubusercontent.com/5550175/59136434-01083600-8938-11e9-8fde-39b290c2a314.png)

Here are some options I have in mind (please chime in if there are others ):
1. We update these styles
2. We add new "templates for "softer" styles if we want to keep existing info alert "loud" styles